module(tests.error-overlapping-pattern)
import(std.prelude)

data(Foo,
    A,
    B)
#

def(f, Foo --,
    match(A -> id, B -> id))

def(g, Foo --,
    match(A -> id, A -> id, B -> id))

def(h, Foo --,
    match(_ -> drop, A -> id))

def(main, ,)
# mirth-test # merr # 13:20: error: Case is unreachable.
# mirth-test # merr # 16:22: error: Case is unreachable.
# mirth-test # mret # 1
